File: DataRepo.Web/src/app/core/models/post.ts
export class Post {
    author: string;
    authorId: number;
    id: number;
    likes: number;
    popularity: number;
    reads: number;
    tags: string[];

    constructor(data: Post) {
        this.author = data.author;
        this.authorId = data.authorId;
        this.id = data.id;
        this.likes = data.likes;
        this.popularity = data.popularity;
        this.reads = data.reads;
        this.tags = data.tags;
    }
}

File: DataRepo/Models/Post.cs
namespace DataRepo.Models;

public class Post
{
    public string? author { get; set; }
    public long authorId {get; set; }
    public long id { get; set; }
    public int likes { get; set; }
    public decimal popularity { get; set; }
    public int reads { get; set; }
    public List<string> tags { get; set; } = new List<string>();
}
